jenkinsbuilder/cccp_index_reader.py

```diff
@@ -110,15 +112,25 @@ def main(indexdlocation):
                                   ':'.join(
                                       [jjb_defaults_file, generated_filename])
                                   ]
+                        if mock:
+                            ret_data.append(
+                                {
+                                    "project_info": new_proj,
+                                    "dump_file": generated_filename,
+                                    "cmd": myargs
+                                }
+                            )
                         print myargs
-                        proc = subprocess.Popen(myargs,
-                                                stdout=subprocess.PIPE)
-                        proc.communicate()
+                        if not mock:
+                            proc = subprocess.Popen(myargs,
+                                                    stdout=subprocess.PIPE)
+                            proc.communicate()
 
                     finally:
                         print "Removing {}".format(t)
                         # shutil.rmtree(t)
-
+                        if mock:
+                            return ret_data
```

Comment:
I thought of using stubs, but then whats the point of stubbing the function under test, you ussually stub stuff that the function needs, but is in accessible due to n number of reasons, such as not having a fully functional jenkins or openshift for example.